configuration/azure/appconfig/appconfig.go

```diff
@@ -160,32 +160,71 @@ func parseMetadata(meta configuration.Metadata) (metadata, error) {
 func (r *ConfigurationStore) Get(ctx context.Context, req *configuration.GetRequest) (*configuration.GetResponse, error) {
 	keys := req.Keys
 	items := make([]*configuration.Item, 0, len(keys))
-	for _, key := range keys {
-		resp, err := r.client.GetSetting(
-			ctx,
-			key,
-			&azappconfig.GetSettingOptions{
-				Label: to.Ptr("label"),
-			})
-		if err != nil {
+
+	if len(keys) == 0 {
+		var err error
+		if items, err = r.getAll(ctx); err != nil {
 			return nil, err
 		}
-
-		item := &configuration.Item{
-			Metadata: map[string]string{},
+	} else {
+		for _, key := range keys {
+			resp, err := r.client.GetSetting(
+				ctx,
+				key,
+				&azappconfig.GetSettingOptions{
+					Label: to.Ptr("label"),
+				})
+			if err != nil {
+				return nil, err
+			}
+
+			item := &configuration.Item{
+				Metadata: map[string]string{},
+			}
+			item.Key = key
+			item.Value = *resp.Value
+			item.Metadata["label"] = *resp.Label
+
+			items = append(items, item)
```

Comment:
You have allocated `items` with a fixed size, so you shouldn't be using `append()`. A better approach would be:

```go
items := make([]*configuration.Item, 0, len(keys))
i := 0
for _, key := range keys {
	// do work here
	items[i] = item
	i++
}
// this is needed in case we are returning less items than keys
items = items[:i]
```